### PR TITLE
Add str interpolation for %x

### PR DESCRIPTION
**Pull request by S-YOU**
Added str interpolation for %x.

### CODE REVIEW

**Comment by S-YOU** [comment]
Nope, I will do that way.

**Comment by S-YOU** [comment]
Rebased, Please take a look.

File: runtime/str.go [hunk 1]
```diff
@@ -625,11 +625,20 @@ func strInterpolate(f *Frame, format string, values *Tuple) (*Object, *BaseExcep
...
-			if raised != nil {
-				return nil, raised
+			if matches[7] == "d" {
+				s, raised := ToStr(f, i)
+				if raised != nil {
+					return nil, raised
+				}
+				buf.WriteString(s.Value())
+			}
+			if matches[7] == "x" || matches[7] == "X" {
```
Comment (trotterdylan) [inline runtime/str.go]: This can just be an `} else {`

File: runtime/str.go [hunk 2]
```diff
@@ -625,11 +625,20 @@ func strInterpolate(f *Frame, format string, values *Tuple) (*Object, *BaseExcep
...
 			}
-			s, raised := ToStr(f, i)
-			if raised != nil {
-				return nil, raised
+			if matches[7] == "d" {
+				s, raised := ToStr(f, i)
+				if raised != nil {
+					return nil, raised
+				}
+				buf.WriteString(s.Value())
```
Comment (trotterdylan) [inline runtime/str.go]: Would be nice to have a single buf.WriteString() call to accompany valueIndex++. Maybe declare a `var val string` and then populate it in each conditional branch.

Comment (S-YOU) [inline runtime/str.go]: declaring var is on the top of func or inside case is ok?

File: runtime/str.go (hunk 1 above)
Comment (S-YOU) [inline runtime/str.go]: Ok, sure.

File: runtime/str.go (hunk 2 above)
Comment (trotterdylan) [inline runtime/str.go]: Inside the case is ok

**Comment by trotterdylan** [comment]
Looks good! Thanks for doing this.

**Pull request merged**